src/SME.SR.Application/Queries/RelatorioSondagemComponentes/ObterRelatorioSondagemComponentes/ObterRelatorioSondagemComponentesPorTurmaQueryHandler.cs

```diff
@@ -0,0 +1,150 @@
+        {
+            return Task.FromResult(new RelatorioSondagemComponentesPorTurmaRelatorioDto() { 
+                Cabecalho = ObterCabecalho(request),
+                Planilha = ObterPlanilha(request)
+            });
+        }
+
+        private RelatorioSondagemComponentesPorTurmaCabecalhoDto ObterCabecalho(ObterRelatorioSondagemComponentesPorTurmaQuery request)
+        {
+            return new RelatorioSondagemComponentesPorTurmaCabecalhoDto()
+            {
+                Ano = request.AnoLetivo,
+                AnoLetivo = request.AnoLetivo,
+                ComponenteCurricular = ObterComponenteCurricular(request.ComponenteCurricularId).Descricao,
+                DataSolicitacao = DateTime.Now,
+                Dre = this.dreRepository.ObterPorCodigo(request.DreCodigo).Result.Abreviacao,
+                Periodo = request.Semestre.ToString(),
+                Proficiencia = "Campo Aditivo",
+                Turma = request.TurmaCodigo,
+                Ue = ObterUe(request.UeCodigo).NomeRelatorio,
+                Rf = request.UsuarioRF,
+                Usuario = this.usuarioRepository.ObterDados(request.UsuarioRF).Result.Login,
+                Ordens = this.relatorioSondagemComponentePorTurmaRepository.ObterOrdens(),
+                Perguntas = ObterPerguntas()
+            };
+        }
+
+        private ComponenteCurricularSondagem ObterComponenteCurricular(string componenteCurricularId)
+        {
+            return this.componenteCurricularRepository.ObterComponenteCurricularDeSondagemPorId(componenteCurricularId).Result.FirstOrDefault();
+
+        }
+
+        private Ue ObterUe(string ueCodigo)
+        {
+            return this.ueRepository.ObterPorCodigo(ueCodigo).Result;
+        }
+
+        public List<RelatorioSondagemComponentesPorTurmaPerguntaDto> ObterPerguntas()
+        {
+            return new List<RelatorioSondagemComponentesPorTurmaPerguntaDto>()
+                {
+                    new RelatorioSondagemComponentesPorTurmaPerguntaDto()
+                    {
+                        Id = 1,
+                        Nome = "Ideia"
+                    },
+                    new RelatorioSondagemComponentesPorTurmaPerguntaDto()
+                    {
+                        Id = 2,
+                        Nome = "Resultado"
+                    }
+                };
+        }
+
+        public RelatorioSondagemComponentesPorTurmaPlanilhaDto ObterPlanilha(ObterRelatorioSondagemComponentesPorTurmaQuery request)
+        {
+            List<RelatorioSondagemComponentesPorTurmaPlanilhaLinhasDto> linhasPlanilhaQueryDto = new List<RelatorioSondagemComponentesPorTurmaPlanilhaLinhasDto>();
+
+            foreach (var linha in this.relatorioSondagemComponentePorTurmaRepository.ObterPlanilhaLinhas(request.DreCodigo, request.TurmaCodigo, request.AnoLetivo, request.Semestre))
```

Comment:
externar para um objeto a lista para fazer o for; Obter o mesmo por await ( tornar o metodo async )